plugins/woocommerce/tests/php/src/Admin/API/MarketingCampaignsTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Test the API controller class that handles the marketing campaigns REST response.
+ *
+ * @package WooCommerce\Admin\Tests\Admin\API
+ */
+
+namespace Automattic\WooCommerce\Tests\Admin\API;
+
+use Automattic\WooCommerce\Admin\Marketing\MarketingCampaign;
+use Automattic\WooCommerce\Admin\Marketing\MarketingChannelInterface;
+use Automattic\WooCommerce\Admin\Marketing\MarketingChannels as MarketingChannelsService;
+use WC_REST_Unit_Test_Case;
+use WP_REST_Request;
+
+/**
+ * MarketingCampaigns API controller test.
+ *
+ * @class MarketingCampaignsTest.
+ */
+class MarketingCampaignsTest extends WC_REST_Unit_Test_Case {
+	/**
+	 * Endpoint.
+	 *
+	 * @var string
+	 */
+	const ENDPOINT = '/wc-admin/marketing/campaigns';
+
+	/**
+	 * @var MarketingChannelsService
+	 */
+	private $marketing_channels_service;
+
+	/**
+	 * Set up.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		// Register an administrator user and log in.
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+		wp_set_current_user( $this->user );
+
+		$this->marketing_channels_service = wc_get_container()->get( MarketingChannelsService::class );
+	}
+
+	/**
+	 * Test teardown.
+	 */
+	public function tearDown(): void {
+		$this->marketing_channels_service->unregister_all();
+		parent::tearDown();
+	}
+
+	/**
+	 * Tests that the marketing campaigns for all registered channels are aggregated and returned by the endpoint.
+	 */
+	public function test_returns_aggregated_marketing_campaigns() {
+		// Create a mock marketing channel.
+		$test_channel_1 = $this->createMock( MarketingChannelInterface::class );
+		$test_channel_1->expects( $this->any() )->method( 'get_slug' )->willReturn( 'test-channel-1' );
+		// Create a mock marketing campaign.
+		$test_campaign_1 = $this->createMock( MarketingCampaign::class );
+		$test_campaign_1->expects( $this->any() )->method( 'get_id' )->willReturn( 'test-campaign-1' );
+		$test_campaign_1->expects( $this->any() )->method( 'get_channel' )->willReturn( $test_channel_1 );
```

Comment:
The `MarketingCampaign::get_channel` method doesn't exist anymore (8ef6532a078153e3bdb196f52cfa8eeb095a7b2b). Looks like it would need to be changed to `get_type`, which would have to return a mocked `MarketingCampaignType`.

This doesn't seem to affect the assertions in this test, though.